From 6106acf7de64501750ff5952f41327472b9cf52b Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Wed, 23 Jan 2019 14:38:26 +0200
Subject: [PATCH 0908/1051] net: mvpp2: recycle check with extended zerocopy
 flag

Strict "skb is recyclable" checking with extended two-bit flag
 SKBTX_ZEROCOPY_FRAG instead of single-bit SKBTX_DEV_ZEROCOPY.

Change-Id: I30fe625b30761777066c6992f8099fde41085216
Signed-off-by: Yan Markman <ymarkman@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/2558
Reviewed-by: Yan Markman <Yan.Markman@cavium.com>
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
Tested-by: sa_ip-sw-jenkins
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
index 8a86afb42a07..dbc7f2725e1a 100644
--- a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
@@ -3298,7 +3298,7 @@ static int mvpp2_recycle_get_bm_id(struct sk_buff *skb)
 	/* ipsec: sp/secpath, _skb_refdst ... */
 	if (!skb_irq_freeable(skb))
 		return -1;
-	if (skb_shinfo(skb)->tx_flags & SKBTX_DEV_ZEROCOPY)
+	if (skb_shinfo(skb)->tx_flags & SKBTX_ZEROCOPY_FRAG)
 		return -1;
 
 	/* Get bm-pool-id */
-- 
2.17.1

